### PR TITLE
Solved the issue 134

### PR DESCRIPTION
**Pull request by teodor994**
- Done solving the issue 134 : https://github.com/cs-pub-ro/operating-systems/issues/134
- Added the generate_skels.py infrastructure to generate the skeletons from the solutions
- Added checkers for the tasks which needed one
- Updated the README.md for every task with useful information on how to generate the skels and run the checker
- Added `solution`

### CODE REVIEW

**Review by teodutu — CHANGES_REQUESTED** [review]
Thanks for the contribution. Most of the checkers are working fine. I made a bunch of inline comments. Address them and then below is some more general feedback.

- Update the paths used by the tester scripts and Makefiles from `../` to `../src/`
- Tasks such as `aslr` and `stack-protector` require changing the Makefiles. Add `TODO`s to them. Ideally update `generate_skels.py` to generate support Makefiles from those in `solution/`.
- Fix the CI/CD failures [1]. Spellcheck sometimes outputs false positives caused by words missing from these lists [2]. To fix those, make a PR to add those missing words this repo [2] (use the makefile in [2] to keep the lists sorted).
- Change your commit message to (according to the contributing guidelines [1] ):

```
chapters/data: Add checkers for `memory-security` tasks

<the description of your PR>

Resolves #134

<your signed-off-by line>
```

[1] https://github.com/cs-pub-ro/operating-systems/pull/149/checks
[2] https://github.com/open-education-hub/actions/tree/main/spellcheck/config
[3] https://github.com/cs-pub-ro/operating-systems/blob/main/CONTRIBUTING.md#commits

**Comment by teodutu** [comment]
You didn't overwrite your first commit (`git commit --amend`), but created a new onw instead. Now do this to squash them together into a single one:
1. `git rebase -i main`
2. This will open your default editor and you'll see your 2 commits. Replace the first word "pick" with "s" (from "squash") on the second line, then save and exit.
3. A new editor window will open with the 2 commit messages cincatenated, allowing you to modify them. Only keep the latter. And change "Rsolves #134" to "Fixes #134" as I didn't see the issue linked with this PR. Maybe Github doesn't support "Resolves" like its docs say [1]?
4. `git push -f`

[1] https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by Alex-deVis** [comment]
Instead of following the premium tutorial wrote by @teodutu (which he rewrites everytime in a different way), you could follow this guide [1]. The gif shows you a clear example of how to do a `git rebase -i`.

[1] https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec

File: chapters/data/memory-security/drills/tasks/aslr/solution/tests/Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ check: $(SHELLCODES)
 	make -C $(FULL_SRC_PATH) clean
 	make clean
 	make -i SRC_PATH=$(FULL_SRC_PATH)
-	./run_all_tests.sh
+	sudo bash ./run_all_tests.sh
```
Comment (teodutu) [inline chapters/data/memory-security/drills/tasks/aslr/solution/tests/Makefile:25]: Why do you need `sudo`?

File: chapters/data/memory-security/drills/tasks/bypassing-stack-protector/solution/tests/Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	nasm -o $@ $<
+
+src:
+	make -C $(FULL_SRC_PATH)
+
+check: $(SHELLCODES)
+	make -C $(FULL_SRC_PATH) clean
+	make clean
+	make -i SRC_PATH=$(FULL_SRC_PATH)
+	sudo bash ./run_all_tests.sh
```
Comment (teodutu) [inline chapters/data/memory-security/drills/tasks/bypassing-stack-protector/solution/tests/Makefile:25]: Again: why `sudo bash` and not just `./run_all_tests.sh`?

File: chapters/data/memory-security/drills/tasks/stack-protector/solution/tests/Makefile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	nasm -o $@ $<
+
+src:
+	make -C $(FULL_SRC_PATH)
+
+check: $(SHELLCODES)
+	make -C $(FULL_SRC_PATH) clean
+	make clean
+	make -i SRC_PATH=$(FULL_SRC_PATH)
+	sudo bash ./run_all_tests.sh
```
Comment (teodutu) [inline chapters/data/memory-security/drills/tasks/stack-protector/solution/tests/Makefile:25]: Ditto: you can just run `./run_all_tests.sh`.

**Review by teodutu — COMMENTED** [review]
The checkers are OK. My only questions are why are you running `sudo bash ./run_all_tests.sh` instead of just `./run_all_tests.sh` in `make check`.

**Review by teodutu — APPROVED** [review]
The content is fine. Great job! Fix the linters and then I'll merge this.

**Comment by teodutu** [comment]
You need to rebase the `main` branch to remove the merge conflicts.

**Comment by teodutu** [comment]
@Alex-deVis can you figure out what's the conflict here?

**Comment by Alex-deVis** [comment]
I'll take a look in 30m

**Pull request merged**